diff --git a/configure.ac b/configure.ac
index e5fb7e7b0b..8959ccb0a1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -932,7 +932,7 @@ if $CPP $CPPFLAGS conftest.c >conftest.out 2>/dev/null; then
   AC_MSG_RESULT([$_arm_arch])
   if test "$_arm_arch" = 7; then
     BASECFLAGS="${BASECFLAGS} -mfloat-abi=softfp -mfpu=vfpv3-d16"
-    LDFLAGS="${LDFLAGS} -march=armv7-a -Wl,--fix-cortex-a8"
+    LDFLAGS="${LDFLAGS} -march=armv7-a"
   fi
 else
   AC_MSG_RESULT([not Android])
